m.neural_network.preparetraining/m.neural_network.preparetraining.py

```diff
@@ -0,0 +1,278 @@
+# % guisection: Parameters
+# %end
+
+# %option
+# % key: no_class_value
+# % type: integer
+# % required: yes
+# % multiple: no
+# % answer: 1
+# % label: Expected and output value for the non class of interest areas (inverse of <class_value>)
+# % guisection: Parameters
+# %end
+
+# %option G_OPT_M_NPROCS
+# %end
+
+import atexit
+from multiprocessing import Pool
+import os
+import random
+
+import grass.script as grass
+from grass_gis_helpers.cleanup import general_cleanup
+from grass_gis_helpers.general import set_nprocs
+from grass_gis_helpers.mapset import verify_mapsets
+from grass.pygrass.modules import Module, ParallelModuleQueue
+from osgeo import gdal
+
+
+# initialize global vars
+ID = grass.tempname(8)
+orig_region = None
+rm_dirs = []
+
+def cleanup():
+    general_cleanup(orig_region=orig_region, rm_dirs=rm_dirs)
+
+def check_parallel_errors(queue):
+    for proc_num in range(queue.get_num_run_procs()):
+        proc = queue.get(proc_num)
+        if proc.returncode != 0:
+            # save all stderr to a variable and pass it to a GRASS
+            # exception
+            errmsg = proc.outputs["stderr"].value.strip()
+            grass.fatal(
+                _(f"\nERROR processing <{proc.get_bash()}>: {errmsg}"),
+            )
+
+def check_parallel_warnings(queue):
+    for mod in queue.get_finished_modules():
+        stderr = mod.outputs["stderr"].value.strip()
+        # the WARNING might get translated...
+        if "WARN" in stderr:
+            grass.warning(_(f"\nWARNING processing <{mod.get_bash()}>: {stderr}"))
+
+def get_tile_infos(in_dir, type):
+    all_tiles = []
+    # get all training_tiles and split into training and validation
+    for tile in os.listdir(in_dir):
+        tiledir = os.path.join(in_dir, tile)
+        tiledict = {}
+        if os.path.isdir(tiledir) and tile.startswith("tile_"):
+            tiledict["id"] = tile
+            tiledict["type"] = type
+            tiledict["dop_tif"] = os.path.join(tiledir, f"image_{tile}.tif")
+            tiledict["ndom_tif"] = os.path.join(tiledir, f"ndsm_1_255_{tile}.tif")
+            checklist = [tiledict["dop_tif"], tiledict["ndom_tif"]]
+            if type == "training":
+                tiledict["label_gpkg"] = os.path.join(tiledir, f"label_{tile}.gpkg")
+                checklist.append(tiledict["label_gpkg"])
+
+            # check if they exist
+            for f in checklist:
+                if not os.path.isfile(f):
+                    grass.fatal(_(f"File {f} expected but no found."))
+
+            all_tiles.append(tiledict)
+    return all_tiles
+
+def build_vrts(outdir, dop, ndom, tile_id):
+    src_ds = gdal.Open(dop)
+    band_count = 0
+    if src_ds is not None:
+        band_count = int(src_ds.RasterCount)
+    else:
+        grass.fatal(_(f"File {dop} is empty."))
+    src_ds = None
+
+    vrt_input = []
+    for num in range(1, band_count + 1):
+        band_vrt = os.path.join(outdir, f"{tile_id}_{num}.tif")
```

Comment:
Ich glaube die einzelnen Band-Tifs dürfen nicht in dem outdir liegen. Und es sollte ein VRT sein.

Reply:
TODO neues Output dir mit den vier (fünf) result ordnern --> In den fünften Ordner sollen die einzel Tifs/VRTs